api/server/server.go

```diff
@@ -1137,8 +1137,17 @@ func (s *Server) getContainersByName(version version.Version, w http.ResponseWri
 		return fmt.Errorf("Missing parameter")
 	}
 
+	if err := parseForm(r); err != nil {
+		return err
+	}
+
+	no_err := r.Form.Get("no_err")
 	containerJSON, err := s.daemon.ContainerInspect(vars["name"])
 	if err != nil {
+		if strings.Contains(strings.ToLower(err.Error()), "no such id") && no_err == "1" {
+			w.WriteHeader(http.StatusNoContent)
```

Comment:
You are right, I was thinking something wrong, we can use 404 in this case, after reading https://github.com/docker/docker/issues/10773#issuecomment-105875168, I got your former comments, wrong container name should not consider as error message on daemon side, it's client's fault.

I'll update this tomorrow, thanks.